source/main/physics/Beam.cpp

```diff
@@ -6597,3 +6597,15 @@ bool Beam::inRange(float num, float min, float max)
 {
 	return (num <= max && num >= min);
 }
+
+Vector3 Beam::getNodePosition(int nodeNumber)
+{
+	if(nodeNumber >= 0 && nodeNumber < free_node)
+	{
+		return nodes[nodeNumber].AbsPosition;
+	}
+	else
+	{
+		return Ogre::Vector3();

```

Comment:
I guess this returns (0,0,0) but what if you query for a not existing node and you wrongly believe the node is at (0,0,0) instead of realizing it doesn't exist? Something like (MAX_INT, MAX_INT, MAX_INT) may be better.